src/briefcase/platforms/android/apk.py

```diff
@@ -1 +1,205 @@
+Found Python version {self.python_version_tag}. Android packaging currently requires Python 3.7.
+        """.format(
+                    self=self
+                )
+            )
+
+
+# The way to split this up is that you might start with more
+# stuff externally specified.
+
+
+class ApkCreateCommand(ApkMixin, CreateCommand):
+    description = "Create and populate an Android APK."
+
+    # TODO: Move these two into my app, and out of this pull request!
+    @property
+    def support_package_url(self):
+        "The URL of the support package to use for apps of this type."
+        # Someday this may move into the briefcase-support.org web app, probably
+        # around the time we support multiple Python versions.
+        return "https://github.com/paulproteus/Python-Android-support/releases/download/v0.5/3.7.zip"
+
+    @property
+    def app_template_url(self):
+        "The URL for a cookiecutter repository to use when creating apps"
+        # Add briefcase.toml to the cookiecutter template
+        return "https://github.com/paulproteus/cookiecutter-beeware-android.git"
+
+
+class ApkUpdateCommand(ApkMixin, UpdateCommand):
+    description = "Update an existing Android APK."
+
+
+class ApkBuildCommand(ApkMixin, BuildCommand):
+    description = "Build an Android APK."
+
+    # e.g.
+    # mock self.download_url to always download a sentinel zip
+    # - validate we create the android_sdk path
+    # see xcode tests for mocks of command line tools
+
+    @property
+    def android_sdk_url(self):
+        """The Android SDK URL appropriate to this operating system."""
+        # These URLs have existed since approximately 2017, and they have a built-in autoupdater,
+        # so I expect them to keep working for quite a few years longer.
+        return "https://dl.google.com/android/repository/sdk-tools-{os}-4333796.zip".format(
+            os=self.host_os.lower()
+        )
+
+    def verify_tools(self):
+        super().verify_tools()
+
+        print()
+        print("Ensuring we have the Android SDK...")
+        if not self.android_sdk_path.exists():
+            try:
+                android_sdk_zip_path = self.download_url(
+                    url=self.android_sdk_url,
+                    download_path=Path.home() / ".briefcase" / "tools",
+                )
+            except requests_exceptions.ConnectionError:
+                raise NetworkFailure("downloading Android SDK")
+            with ZipFile(android_sdk_zip_path) as android_sdk_zip:
+                android_sdk_zip.extractall(path=self.android_sdk_path_tmp)
+            # Remove the ZIP file; it has no purpose now that it is extracted.
+            android_sdk_zip_path.unlink()
+            # Set executable permissions; Python ZipFile ignores these, but we need them.
+            for binpath in (self.android_sdk_path_tmp / "tools" / "bin").glob("*"):
+                binpath.chmod(0o755)
+            self.android_sdk_path_tmp.rename(self.android_sdk_path)
```

Comment:
I can see a bunch of possible error cases here that aren't caught (bad zipfile; no permissions to unpack zipfile; disk space problems when unpacking...).